frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts

```diff
@@ -35,11 +36,17 @@ export const hideAllExistingProperties = (schema: JSONSchema7) => {
 const k8sResourceCapabilityToUISchema = (capability: SpecCapability): UiSchema => {
   const [, suffix] = capability.match(REGEXP_K8S_RESOURCE_SUFFIX) ?? [];
   const groupVersionKind = suffix?.replace(/:/g, '~');
+
+  let [, selector] = capability.match(REGEXP_K8S_RESOURCE_SELECTOR_SUFFIX) ?? [];
+  if (selector) {
+    selector = selector.replace(/\+/g, ' ');
+  }
```

Comment:
If we capture all tokens with one regex expression, we should be able to simplify this.
```suggestion
  const [, groupVersionKindToken, selectorToken] = capability.match(REGEXP_K8S_RESOURCE_SUFFIX) ?? [];
  const groupVersionKind = groupVersionKindToken?.replace(/:/g, '~');
  const selector = selectorFromString(selectorToken?.replace(/\+/g, ' '));
```